src/IO/H5/Python/CombineH5DatSegments.py

```diff
@@ -0,0 +1,95 @@
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+
+import logging
+import os
+import shutil
+
+import click
+import h5py
+
+
+def combine_h5_dat_segments(output, input_files):
+    """Combines multiple HDF5 dat files
+
+    This executable is used for combining a series of HDF5 files, each
+    containing one or more dat files, into a single HDF5 file. A typical
+    use case is to join dat-containing HDF5 files from different segments
+    of a simulation, with each segment containing values of the dat files
+    during different time intervals."""
+    if len(input_files) == 0:
+        logging.warning("CombineH5DatSegments: no input files, nothing to do")
+        return
+
+    # Copy first input file to output file
+    output_path = output + ".h5"
+    shutil.copy(input_files[0], output_path)
+
+    # Open the output file for appending
+    with h5py.File(output_path, "r+") as out:
+        # Get a list of all dat file keys (keys ending in ".dat")
+        dat_file_keys = []
+        out.visit(
+            lambda key: (
+                dat_file_keys.append(key)
+                if str(key).split(".")[-1] == "dat"
+                else None
+            )
+        )
+
+        # Loop over remaining input files, appending each dat file
+        for input_file in input_files[1:]:
+            with h5py.File(input_file, "r") as input:
+                for dat_file_key in dat_file_keys:
+                    if dat_file_key in input.keys():
+                        data_to_append = input[dat_file_key]
+                        start_size = out[dat_file_key].shape[0]
+                        append_size = input[dat_file_key].shape[0]
+                        out[dat_file_key].resize(
+                            start_size + append_size, axis=0
+                        )
+                        out[dat_file_key][start_size:] = input[dat_file_key]
+
+
+@click.command(
+    name="combine-h5-dat-segments", help=combine_h5_dat_segments.__doc__
+)
+@click.argument(
+    "input_files",
+    type=click.Path(
+        exists=True,
+        file_okay=True,
+        dir_okay=False,
+        readable=True,
+    ),
+    nargs=-1,
+)
+@click.option(
+    "--output",
+    "-o",
+    required=True,
+    type=click.Path(
+        exists=False,
+        file_okay=True,
+        dir_okay=False,
+        writable=True,
+        readable=True,
+    ),
+    help="Combined output filename (omit file extension).",
+)
+@click.option(
+    "--force",
+    "-f",
+    is_flag=True,
+    help="If the output file already exists, overwrite it.",
+)
+def combine_h5_dat_segments_command(output, input_files, force):
+    # If output file exists, exit unless the user specifies `--force`
+    if os.path.exists(output + ".h5") and not force:
+        raise ValueError(f"File '{output}' exits; to overwrite, use --force")
```

Comment:
Move this to the function above, just before you `shutil.copy`.

You can forward all arguments like this:

```py
def combine_h5_dat_segments_command(**kwargs):
    combine_h5_dat_segments(**kwargs)
```